packages/server/lib/utils/express.ts

```diff
@@ -1,9 +1,31 @@
-import type { DBEnvironment, DBTeam, ConnectSession, DBUser } from '@nangohq/types';
+import type { DBEnvironment, DBTeam, ConnectSession, DBUser, EndUser } from '@nangohq/types';
+
+// Types are historically loose so we need to fix them at some point
+// export type RequestLocals =
+//     | {
+//           authType: 'connectSession';
+//           account: DBTeam;
+//           environment: DBEnvironment;
+//           connectSession: ConnectSession;
+//           endUser: EndUser;
+//       }
+//     | {
+//           authType: 'publicKey';
+//           account: DBTeam;
+//           environment: DBEnvironment;
+//       }
+//     | {
+//           authType: 'basic' | 'session' | 'secretKey' | 'adminKey' | 'none';
+//           account?: DBTeam;
+//           environment?: DBEnvironment;
+//           user: Pick<DBUser, 'id' | 'email' | 'name'>;
+//       };
```

Comment:
Tried to fix this type but opened a trap door (unfortunately I have wrote this type so I can only blame myself)
Will fix later

Reply:
not sure I follow but I trust you

Reply:
This type is too loose meaning we can have connectSession with auth mode basic, which could lead to mistakes. Ideally, we have something stricter but when I wrote it I didn't anticipate this issue and now it seems some middleware does not enjoy the strictness when I tried to change it